### ISSUE TITLE
Learning rate scheduler not tracked in Tensorboard

### ISSUE THREAD

**Issue by KRiedmiller** [keras-team/keras#16087]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.3 LTS
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.8.0
- Python version: 3.9

**Describe the problem**.

When using a learning rate scheduler the learning rate is supposed to be tracked and plotted in the 'Scalars' tab. This, however, does not happen, the plot is simply missing. 

The root of the problem is in the [`tensorflow/python/keras/callbacks.py`](https://github.com/keras-team/keras/blob/master/keras/callbacks.py) file, in the function `_collect_learning_rate`:  
`isinstance(lr_schedule, learning_rate_schedule.LearningRateSchedule)` never evaluates to True.  
As a simple check, I replaced the line with `if callable(lr_schedule)`, which resolves the issue.

Most likely this workaround fails in some situations, but it is not clear to me how to fix that properly.

I imported the schedulers in the following way:
```python
from tensorflow.keras.optimizers.schedules import ExponentialDecay
```

**Describe the current behavior**.
No learning rate plot in Tensorboard

**Describe the expected behavior**.
Learning rate plot in Tensorboard

**Standalone code to reproduce the issue**.

[Simple NN with lr scheduler and tensorboard](https://gist.github.com/KRiedmiller/aaecfb439db3e0017847af97696b226f), mostly taken from official guide.  
Tested with original `tensorflow/python/keras/callbacks.py` and 'fixed' version.

**Comment by gowthamkpr** [comment]
@KRiedmiller Can you please provide code to reproduce this issue. Thanks!

**Comment by KRiedmiller** [comment]
Thank you for looking into this.
If you look at the end of my bug report you find a link to code for testing. With the unfixed `callbacks.py` the tensorboard output does not contain the learning rate, with the 'fixed' `callbacks.py` the provided code produces correct plots of the learning rate.

**Comment by qlzh727** [comment]
The error is caused by using tensorflow.python.keras.callbacks, which is the legacy keras code and about to be deleted. Please use tf.keras.callbacks.TensorBoard instead (not the one under *python*/keras).
